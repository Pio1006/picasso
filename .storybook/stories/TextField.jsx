import React from 'react'
import { storiesOf } from '@storybook/react'
import TextField from '../../components/TextField'
import SearchIcon from '@material-ui/icons/Search'

storiesOf('TextField', module)
  .add('text', () => <TextField label="Search..." />)
  .add('text with Icon', () => (
    <TextField label="Search..." Icon={<SearchIcon />} />
  ))
  .add('select', () => (
    <div style={{ width: '300px' }}>
      <TextField
        label="Gender"
        select
        fullWidth
        SelectProps={{
          native: true
        }}>
        <option value="" />
        <option value="male">Male</option>
        <option value="female">Female</option>
      </TextField>
    </div>
  ))
  .add('textarea', () => (
    <TextField label="Description..." rows={4} multiline />
  ))
