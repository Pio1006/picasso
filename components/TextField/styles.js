import '../InputBase/styles'
import '../InputLabel/styles'
import '../OutlinedInput/styles'

export default {
  TextField: {
    iconStart: {
      transform: 'translate(2.8em, 1em) scale(1)',
      '&$shrink': {
        transform: 'translate(2.8em, 0.6em) scale(.75)'
      }
    },
    shrink: {}
  }
}
