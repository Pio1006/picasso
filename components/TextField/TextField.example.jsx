import React from 'react'
import { storiesOf } from '@storybook/react'
import SearchIcon from '@material-ui/icons/Search'

import TextField from './'

storiesOf('TextField', module)
  .add('text', () => <TextField label='Search...' />)
  .add('error', () => <TextField error label='Search...' />)
  .add('text with Icon', () => (
    <TextField Icon={<SearchIcon />} label='Search...' />
  ))
  .add('select', () => (
    <div style={{ width: '300px' }}>
      <TextField
        fullWidth
        label='Gender'
        select
        // eslint-disable-next-line react/jsx-sort-props
        SelectProps={{
          native: true
        }}
      >
        <option value='' />
        <option value='y'>Yes</option>
        <option value='n'>No</option>
      </TextField>
    </div>
  ))
  .add('textarea', () => (
    <TextField label='Description...' multiline rows={4} />
  ))
  .add('all', () => (
    <div style={{ width: '300px' }}>
      <TextField Icon={<SearchIcon />} fullWidth label='Search...' />
      <div style={{ marginBottom: '1em' }} />
      <TextField
        SelectProps={{
          native: true
        }}
        fullWidth
        label='Gender'
        select
      >
        <option value='' />
        <option value='y'>Yes</option>
        <option value='n'>No</option>
      </TextField>
      <div style={{ marginBottom: '1em' }} />
      <TextField fullWidth label='Description...' multiline rows={4} />
    </div>
  ))
