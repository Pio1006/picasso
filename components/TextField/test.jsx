import React from 'react'
import TextField from './index'
import { render, cleanup } from 'react-testing-library'
import SearchIcon from '@material-ui/icons/Search'

const renderTextField = (children, props = {}) => {
  return render(<TextField {...props}>{children}</TextField>)
}

afterEach(cleanup)

describe('Icon prop', () => {
  test('renders icon at the end', () => {
    const { container } = renderTextField(null, {
      Icon: <SearchIcon />
    })

    expect(container).toMatchSnapshot()
  })

  test('renders icon at the beginning', () => {
    const { container } = renderTextField(null, {
      Icon: <SearchIcon />,
      iconPosition: 'start'
    })

    expect(container).toMatchSnapshot()
  })
})
