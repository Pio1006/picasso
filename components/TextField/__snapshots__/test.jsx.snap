// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon prop renders icon at the beginning 1`] = `
<div>
  <div
    class="MuiFormControl-root-50"
  >
    <div
      class="MuiInputBase-root-67 MuiFilledInput-root-54 MuiFilledInput-underline-55 MuiInputBase-formControl-68 MuiInputBase-adornedStart-71 MuiFilledInput-adornedStart-58"
    >
      <div
        class="MuiInputAdornment-root-84 MuiInputAdornment-positionStart-86"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-88"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiInputBase-input-77 MuiFilledInput-input-62 MuiInputBase-inputAdornedStart-82 MuiFilledInput-inputAdornedStart-65"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Icon prop renders icon at the end 1`] = `
<div>
  <div
    class="MuiFormControl-root-2"
  >
    <div
      class="MuiInputBase-root-19 MuiFilledInput-root-6 MuiFilledInput-underline-7 MuiInputBase-formControl-20 MuiInputBase-adornedEnd-24 MuiFilledInput-adornedEnd-11"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input-29 MuiFilledInput-input-14 MuiInputBase-inputAdornedEnd-35 MuiFilledInput-inputAdornedEnd-18"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root-36 MuiInputAdornment-positionStart-38"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-40"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
